### ISSUE TITLE
Question about running FQF agent on Breakout

### ISSUE THREAD

**Issue by Bowen-He** [ku2482/fqf-iqn-qrdqn.pytorch#12]
Hi, I ran the FQF agent on Breakout, but just turned out to get a curve that collapses in the middle(like in the attachment). 
![picture](https://user-images.githubusercontent.com/71317990/106474281-b8739800-64df-11eb-9006-f6be754fca20.PNG)
The command I was using is 
python3 -u train_fqf.py --cuda --env_id BreakoutNoFrameskip-v4 --seed 0 --config config/fqf.yaml .
And the hyperparameters is the default ones.
Should I adjust some hyperparameters to get a curve that you have achieved?

**Comment by toshikwa** [comment]
@Bowen-He 

In a personal communication, the author mentions the following. Your problem seems to be related to it.
> We did not use the regularization term in our implementation. The reason why we mention this term is because that in some rare case (about 1 out of 20 seeds), the proposed fraction may degenerates into a deterministic one (e.g. [0,0,0,...,1,0,...]).

In FQF paper, the author proposed to use the entropy bonus (regularization term) to prevent it.
I implemented it so that you can use the entropy bonus setting `ent_coef` bigger than 0 in the [config](https://github.com/ku2482/fqf-iqn-qrdqn.pytorch/blob/master/config/fqf.yaml#L5).
(I haven't tested it due to the limited resources.)

Thanks.

**Comment by Bowen-He** [comment]
@ ku2482
Thank you! I'll do experiments immediately, and see if there is any change. I've done with seed 0 and seed 5, but they all fall into the same situation eventually. Hope a regularizer could help!

**Comment by Bowen-He** [comment]
@ku2482 
Thank you! I'll do experiments immediately, and see if there is any change. I've done with seed 0 and seed 5, but they all fall into the same situation eventually. Hope a regularizer could help!

**Comment by Bowen-He** [comment]
@ku2482 
Hi! I used the entropy loss, and code just runs well. And I still have a question about the returns. Could you please tell me why the train return is so far from the test return. I checked train_episode() and evaluate() in base_agent.py, but I couldn't get the explaination here.
![捕获](https://user-images.githubusercontent.com/71317990/107180502-c8c1d080-6a13-11eb-9b45-ad77c09edaca.PNG)

**Comment by toshikwa** [comment]
@Bowen-He 
Thank you for sharing the result ;)

We clip the reward when training.
https://github.com/ku2482/fqf-iqn-qrdqn.pytorch/blob/master/fqf_iqn_qrdqn/env.py#L166
